File: src/guide/models/shared.py
from enum import Enum


class GameSystem(str, Enum):
    dnd5e = "dnd5e"
    pathfinder2e = "pathfinder2e"


class CharacterType(str, Enum):
    pc = "pc"
    npc = "npc"
    monster = "monster"


class Condition(str, Enum):
    blinded = "blinded"
    charmed = "charmed"
    deafened = "deafened"
    frightened = "frightened"
    grappled = "grappled"
    incapacitated = "incapacitated"
    invisible = "invisible"
    paralyzed = "paralyzed"
    petrified = "petrified"
    poisoned = "poisoned"
    prone = "prone"
    restrained = "restrained"
    stunned = "stunned"
    unconscious = "unconscious"


class EncounterStatus(str, Enum):
    pending = "pending"
    active = "active"
    completed = "completed"
    fled = "fled"


class LoreType(str, Enum):
    npc = "npc"
    location = "location"
    item = "item"
    plot = "plot"
    mechanic = "mechanic"
    backstory = "backstory"
    session_event = "session_event"


class EventType(str, Enum):
    combat = "combat"
    exploration = "exploration"
    social = "social"
    rest = "rest"
    level_up = "level_up"
    item_found = "item_found"
    npc_met = "npc_met"
    plot_revealed = "plot_revealed"
    custom = "custom"


class EventSignificance(str, Enum):
    minor = "minor"
    major = "major"
    milestone = "milestone"


class IngestionStatus(str, Enum):
    pending = "pending"
    processing = "processing"
    completed = "completed"
    failed = "failed"


class Perspective(str, Enum):
    dm = "dm"
    player = "player"

File: crates/guide-core/src/models/shared.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, Default, utoipa::ToSchema)]
#[serde(rename_all = "snake_case")]
pub enum GameSystem {
    #[default]
    Dnd5e,
    Pathfinder2e,
}

#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, utoipa::ToSchema)]
#[serde(rename_all = "snake_case")]
pub enum CharacterType {
    Pc,
    Npc,
    Monster,
}

/// D&D 5e conditions (PHB p. 290-293)
#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, utoipa::ToSchema)]
#[serde(rename_all = "snake_case")]
pub enum Condition {
    Blinded,
    Charmed,
    Deafened,
    Frightened,
    Grappled,
    Incapacitated,
    Invisible,
    Paralyzed,
    Petrified,
    Poisoned,
    Prone,
    Restrained,
    Stunned,
    Unconscious,
}

#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, Default, utoipa::ToSchema)]
#[serde(rename_all = "snake_case")]
pub enum EncounterStatus {
    #[default]
    Pending,
    Active,
    Completed,
    Fled,
}

#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, utoipa::ToSchema)]
#[serde(rename_all = "snake_case")]
pub enum LoreType {
    Npc,
    Location,
    Item,
    Plot,
    Mechanic,
    Backstory,
    SessionEvent,
}

#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, utoipa::ToSchema)]
#[serde(rename_all = "snake_case")]
pub enum EventType {
    Combat,
    Exploration,
    Social,
    Rest,
    LevelUp,
    ItemFound,
    NpcMet,
    PlotRevealed,
    Custom,
}

#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, Default, utoipa::ToSchema)]
#[serde(rename_all = "snake_case")]
pub enum EventSignificance {
    #[default]
    Minor,
    Major,
    Milestone,
}

#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, Default, utoipa::ToSchema)]
#[serde(rename_all = "snake_case")]
pub enum IngestionStatus {
    #[default]
    Pending,
    Processing,
    Completed,
    Failed,
}

/// Who is viewing the information — controls spoiler filtering.
#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, utoipa::ToSchema)]
#[serde(rename_all = "snake_case")]
pub enum Perspective {
    Dm,
    Player,
}
